etk-asm/src/asm.rs

```diff
@@ -430,203 +446,48 @@ impl Assembler {
                         }
                         .fail()
                     }
-                    Err(_) => {
-                        assert_eq!(self.pending_len, Some(0));
-                        self.pending_len = rop.size();
-                        self.pending.push_back(rop);
+                    Err(ops::Error::ContextIncomplete {
+                        source: UnknownLabel { label: _label, .. },
+                    }) => {
+                        let mut dynamic_push = false;
+                        match op.size() {
```

Comment:
Why do you use `size` to determine if this is a variable push? A more accurate way would be something like:

```rust
if let AbstractOp::Push(_) = op { ... }
```

Reply:
My idea was to make it as general as possible. I was not only interested in pushes, but in any other operation that won't have a defined size in the future. But rethinking it, you'll probably have to modify other parts of the code if that happens, so the idea of using if let AbstractOp::Push(_) makes a lot of sense. Thanks!